core/core-impl/src/main/java/ai/timefold/solver/core/config/solver/SolverConfigOverride.java

```diff
@@ -0,0 +1,38 @@
+package ai.timefold.solver.core.config.solver;
+
+import ai.timefold.solver.core.api.domain.solution.PlanningSolution;
+import ai.timefold.solver.core.config.solver.termination.TerminationConfig;
+
+/**
+ * Includes settings to override default {@link ai.timefold.solver.core.api.solver.Solver} configuration.
+ *
+ * @param <Solution_> the solution type, the class with the {@link PlanningSolution} annotation
+ */
+public class SolverConfigOverride<Solution_> {
```

Comment:
It is questionable if this class belongs in `config` package. As you can see, it caused it to be needlessly inserted in the XSD. I'd move this to `api/solver`, which is where it is primarily used.

Also, let's make it `final`. I am a big friend of removing freedom from people; if you allow people to do something, in this case extending the class, they will. And then you have to live with that forever.

Reply:
> It is questionable if this class belongs in `config` package. As you can see, it caused it to be needlessly inserted in the XSD. I'd move this to `api/solver`, which is where it is primarily used.
> 

Yes, I was unsure about how to deal with this XSD auto-generation, and it makes sense to move it to `api/solver.`

> Also, let's make it `final`. I am a big friend of removing freedom from people (in code); if you allow people to do something, in this case extending the class, they will. And then you have to live with that forever.
> 
> We can always allow it if we see the need in the future. We can never forbid it once we've allowed it.

Done!